models/servers.go

```diff
@@ -0,0 +1,100 @@
+package models
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+)
+
+// Server maps AsyncAPI serves.item object
+type Server struct {
+	Extensions    Extensions       `json:"-"`
+	URL           string           `json:"url"`
+	Description   string           `json:"description,omitempty"`
+	Scheme        string           `json:"scheme"`
+	SchemeVersion string           `json:"schemeVersion,omitempty"`
+	Variables     *ServerVariables `json:"variables,omitempty"`
+	BaseChannel   string           `json:"baseChannel,omitempty"`
+	// Security      []*SecurityRequirement `json:"security,omitempty"`
+}
+
+// Unmarshal unmarshals JSON
+func (value *Server) Unmarshal(data []byte) error {
+	return json.Unmarshal(data, value)
+}
+
+// Marshal marshals JSON
+func (value *Server) Marshal() ([]byte, error) {
+	if value.Extensions == nil {
+		return json.Marshal(value)
+	}
+	return MarshalWithExtensions(value, value.Extensions)
+}
```

Comment:
We should avoid calling `json.Marshal(value)` inside the Marshal method. I know I've done it in the Info object but now we know there's the other way of doing it. This method should just create a map, call `json.Marshal(theNewMap)`, and return the resulting byte array. If we do it and also rename the method to `MarshalJSON` the `json.Marshal` function will automatically call it.